source/cl/test/UnitCL/source/ktst_regression_03.cpp

```diff
@@ -609,10 +609,18 @@ struct StrideMisaligned {
   cl_uint work_dim;
 
   bool operator==(const StrideMisaligned &rhs) {
-    return memcmp(this, &rhs, sizeof(StrideMisaligned)) == 0;
+    // clang-tidy does not realise that StrideMisaligned has unique object
+    // representations because of its global_size member: clang has an open
+    // FIXME for supporting vectors. The static_assert below should cover any
+    // scenario where the compiler still adds padding inside the struct, so we
+    // can suppress the clang-tidy warning.
+    return memcmp  // NOLINT(bugprone-suspicious-memory-comparison,cert-exp42-c,cert-flp37-c)
+           (this, &rhs, sizeof(StrideMisaligned)) == 0;
   }
 };
 
+static_assert(sizeof(StrideMisaligned) == sizeof(cl_ulong4) + sizeof(cl_uint));
```

Comment:
if sizeof(StrideMisaligned) is wrong because sizeof(cl_ulong4) was wrong we may still see issues, I wonder if it's better to do sizeof(cl_ulong)*4 + sizeof(cl_uint).
I'm fine with this though if you want to leave it.

Reply:
I think you're right, that's probably better. Will update.

Reply:
Done, https://github.com/codeplaysoftware/oneapi-construction-kit/compare/bcd8d5649c6daf6cf11645929b6813162255e3de..524de2d5147c33a57b0e9132b5414c8f57d072a4. As I expect to now merge this as is I have also done the squashing, but included the diff link here for easier review.